lib/promoted/ruby/client.rb

```diff
@@ -168,7 +168,8 @@ def deliver args, headers={}
             log_request = {
               :full_insertion => response_insertions,
               :experiment => cohort_membership_to_log,
-              :request => request_to_log
+              :request => request_to_log,
+              :client_request_id => delivery_request_builder.client_request_id
```

Comment:
client_request_id is a field on Request, not LogRequest.  Ditto down below.

Reply:
Generally speaking, this is one of the more challenging parts of programming against LogRequest especially, namely, what is going on at the top level vs. what is going on in the nested "request" field.

I think there might also be some misunderstanding of how the Ruby code works. The structure of the final messages that get send to Metrics API and Delivery API respectively are built in RequestBuilder.log_request_params and RequestBuilder.delivery_request_params. Hop over there and check it out, but, in each case, client_request_id ends up at the top level. (You can also see this in the diff of request_builder_spec.rb.)

Can you reconfirm that this is not correct, and I guess in that case we expect client_request_id to be on the _inner_ request (that only gets filled in when !insertions_from_delivery. Thanks.

Reply:
Updated per our conversation, please look again (especially the test cases that show the usage). Thanks!